Lib/dataclasses.py

```diff
@@ -1326,10 +1347,27 @@ def _add_slots(cls, is_frozen, weakref_slot, defined_fields):
             if _update_func_cell_for__class__(member, cls, newcls):
                 break
         elif isinstance(member, property):
-            if (_update_func_cell_for__class__(member.fget, cls, newcls)
-                or _update_func_cell_for__class__(member.fset, cls, newcls)
-                or _update_func_cell_for__class__(member.fdel, cls, newcls)):
-                break
+            for f in member.fget, member.fset, member.fdel:
+                if f is None:
+                    continue
+                # unwrap once more in case function
+                # was wrapped before it became property
+                f = inspect.unwrap(f)
+                if _update_func_cell_for__class__(f, cls, newcls):
+                    break
```

Comment:
Doesn't this `break` only return from the inner loop, but you want it to return from the outer loop? This is the reason I used the `or` clauses in the original. Well that and I didn't want to create the tuple for the inner loop to iterate over, but that's probably an unneeded micro optimization.

Reply:
Good catch. 

One way to fix this would to use this monstrosity: 
```suggestion
                    break
            else:
                continue
            break
```

Regarding `or` clause, I figured it would be slightly faster than the whole iteration thing, but since there's now unwrapping here, went with the loop for sake of less code duplication.

Reply:
Yeah, not a lot of great options here. I think I'd leave the `or` version and add the 3 calls to `unwrap`. Which I realize might make the lines too long, but I think it's better than the above "monstrosity".

Reply:
Just had another idea how we can handle this: https://github.com/python/cpython/pull/124692/commits/4941a3fdb2d1f23c55b8d45968a78dc2e074ebd1.

Reply:
Good one!